src/gtest/test_libzendoo.cpp

```diff
@@ -1973,4 +1982,140 @@ TEST(CctpLibrary, TestInvalidProofVkWhenOversized)
     EXPECT_FALSE(vkInvalid.IsValid());
 
     //TODO: Might be useful to test the same behaviour with bit vector
-}
\ No newline at end of file
+}
+
+TEST(CctpLibrary, TestRandomCustomFieldGeneration)
+{
+    srand(RANDOM_CUSTOM_FIELD_SEED);
+
+    for (int i = 0; i < 32; i++)
+    {
+        ASSERT_EQ(rand() % 256, RANDOM_CUSTOM_FIELD[i]);
+    }
+}
+
+TEST(CctpLibrary, TestGetLeadingZeros)
+{
+    ASSERT_EQ(8, getLeadingZeroBitsInByte(0));
+
+    for (uint8_t n = 1; n > 0; n++)
+    {
+        ASSERT_EQ(__builtin_clz(n) % (sizeof(unsigned int) * 8 - CHAR_BIT), getLeadingZeroBitsInByte(n));
+    }
+}
+
```

Comment:
Could be useful also add a unit test for the getBytesFromBits() api in util.cpp

Reply:
Done with commit [304694d2ecefdfe23ee5750d1c44e1b4562ca5ba](https://github.com/HorizenOfficial/zen/pull/429/commits/304694d2ecefdfe23ee5750d1c44e1b4562ca5ba).